### PR TITLE
CI: Update CUDA, MOFED & CI image

### PR DESCRIPTION
**Pull request by B-a-S**
## What
Migrated to NGC pytorch image 

## Why ?
Updated CUDA & Mofed, using the image that active develop

## How ?
NGC pytorch is  the newest image included PyTorch & CUDA 12.3.
Added MOFED 23.10-0.5.5.0

### CODE REVIEW

File: .ci/Dockerfile.ngc_pytorch [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ARG CUDA_VER='11.4.2'
```
Comment (artemry-nv) [inline .ci/Dockerfile.ngc_pytorch:1]: Let's change the default value to something more actual

File: .ci/Dockerfile.ngc_pytorch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+FROM harbor.mellanox.com/torch-ucc/ucc/1.0.0/x86_64/centos8/cuda${CUDA_VER}:base
```
Comment (artemry-nv) [inline .ci/Dockerfile.ngc_pytorch:2]: Is it still in use?

File: .ci/Dockerfile.ngc_pytorch [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+#FROM nvcr.io/nvidia/pytorch:23.10-py3
```
Comment (artemry-nv) [inline .ci/Dockerfile.ngc_pytorch:3]: Remove unused commented code

File: .ci/build_base_docker/Dockerfile.ngc_pytorch.base [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+FROM nvcr.io/nvidia/pytorch:23.11-py3
+#==============================================================================
+ARG NVIDIA_ROOT_DIR=/opt/nvidia
+ENV DEBIAN_FRONTEND=noninteractive
+ENV TZ=Etc/UTC
+ENV SRC_DIR=${NVIDIA_ROOT_DIR}/src
+ENV PKG_DIR=${NVIDIA_ROOT_DIR}/pkg
+ENV BIN_DIR=${NVIDIA_ROOT_DIR}/bin
+ENV WORKLOADS_DIR=${NVIDIA_ROOT_DIR}/workloads
+ENV TORCH_UCC_GITHUB_URL=https://github.com/facebookresearch/torch_ucc.git
```
Comment (artemry-nv) [inline .ci/build_base_docker/Dockerfile.ngc_pytorch.base:11]: Remove if unused

File: .ci/build_base_docker/Dockerfile.ngc_pytorch.base [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+ENV TORCH_UCC_BRANCH=main
```
Comment (artemry-nv) [inline .ci/build_base_docker/Dockerfile.ngc_pytorch.base:12]: Remove if unused

File: .ci/job_matrix.yaml [hunk 6]
```diff
@@ -77,7 +77,7 @@ steps:
     run: |
       export UCC_PASSWORD=$UCC_PASSWORD
       export UCC_USERNAME=$UCC_USERNAME
-      echo "Running coverity"
+      echo "Running coverity "
```
Comment (artemry-nv) [inline .ci/job_matrix.yaml:80]: Unnecessary changes?

File: .ci/Dockerfile.ngc_pytorch (hunk 1 above)
Comment (artemry-nv) [inline .ci/Dockerfile.ngc_pytorch:1]: Do we need CUDA_VER var at all?

**Review by amaslenn — APPROVED** [review]

File: .ci/Dockerfile.ngc_pytorch (hunk 3 above)
Comment (B-a-S) [reply .ci/Dockerfile.ngc_pytorch]: changed

File: .ci/build_base_docker/Dockerfile.ngc_pytorch.base (hunk 4 above)
Comment (B-a-S) [reply .ci/build_base_docker/Dockerfile.ngc_pytorch.base]: Of course, why not?

File: .ci/build_base_docker/Dockerfile.ngc_pytorch.base (hunk 5 above)
Comment (B-a-S) [reply .ci/build_base_docker/Dockerfile.ngc_pytorch.base]: removed

File: .ci/job_matrix.yaml (hunk 6 above)
Comment (B-a-S) [reply .ci/job_matrix.yaml]: Fixed

File: .ci/Dockerfile.ngc_pytorch (hunk 1 above)
Comment (B-a-S) [reply .ci/Dockerfile.ngc_pytorch]: I think so

File: .ci/Dockerfile.ngc_pytorch (hunk 2 above)
Comment (artemry-nv) [reply .ci/Dockerfile.ngc_pytorch:2]: ?

Comment (artemry-nv) [reply .ci/Dockerfile.ngc_pytorch:2]: `ucc/1.0.0` outdated version

**Review by artemry-nv — APPROVED** [review]

File: .ci/Dockerfile.ngc_pytorch [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+COPY . ${SRC_DIR}/ucc
+
+RUN apt update && apt install -y sudo && \
+    echo "swx-jenkins ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+RUN pip install 'protobuf<=3.19.0'
+#==============================================================================
+# Build UCC
+RUN ${SRC_DIR}/ucc/.ci/scripts/build_ucc.sh
+#==============================================================================
+# Install torch_ucc (UCC version) python module and build a wheel package
```
Comment (Sergei-Lebedev) [inline .ci/Dockerfile.ngc_pytorch:13]: misleading comment

**Review by Sergei-Lebedev — APPROVED** [review]

**Comment by artemry-nv** [comment]
Can we merge the PR?

File: .ci/Dockerfile.ngc_pytorch (hunk 7 above)
Comment (B-a-S) [reply .ci/Dockerfile.ngc_pytorch:13]: fixed

File: .ci/Dockerfile.ngc_pytorch (hunk 2 above)
Comment (B-a-S) [reply .ci/Dockerfile.ngc_pytorch:2]: It's just a path
